app/src/main/kotlin/compiler/common/diagnostics/Diagnostic.kt

```diff
@@ -32,4 +32,14 @@ sealed class Diagnostic {
                 append(" Expected symbols: ${expectedSymbols.joinToString()}.")
         }.toString()
     }
+
+    sealed class NameResolutionErrors() : Diagnostic() {
+        override fun isError() = true
+
+        class UndefinedVariable : NameResolutionErrors()
```

Comment:
Don't you think that we should print some more info, like in parser/lexer? I.e. name of the variable, line, etc.?